reportsrv/reportsrv.go

```diff
@@ -0,0 +1,194 @@
+package reportsrv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/RMI/pacta/blob"
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/pacta"
+	"github.com/go-chi/chi/v5"
+	"go.uber.org/zap"
+)
+
+type Config struct {
+	DB     DB
+	Blob   Blob
+	Logger *zap.Logger
+}
+
+func (c *Config) validate() error {
+	if c.DB == nil {
+		return errors.New("no DB was given")
+	}
+	if c.Blob == nil {
+		return errors.New("no blob client was given")
+	}
+	if c.Logger == nil {
+		return errors.New("no logger was given")
+	}
+	return nil
+}
+
+type Server struct {
+	db     DB
+	blob   Blob
+	logger *zap.Logger
+}
+
+type DB interface {
+	NoTxn(context.Context) db.Tx
+
+	AnalysisArtifactsForAnalysis(tx db.Tx, id pacta.AnalysisID) ([]*pacta.AnalysisArtifact, error)
+	Blobs(tx db.Tx, ids []pacta.BlobID) (map[pacta.BlobID]*pacta.Blob, error)
```

Comment:
I think you're planning this for the next PR, if so great, but we'll need to generate Audit Logs here.

Reply:
Indeed, will do